src/client.rs

```diff
@@ -98,33 +104,18 @@ impl Tidal {
     }
 
     fn session_id(&self) -> String {
-        let credentials = self.credentials.clone();
-        let session_id = match &credentials.session_info {
-            None => panic!("A session needs to be obtained before using Tidal"),
-            Some(session_info) => match &session_info.session_id {
-                Some(session_id) => session_id,
-                None => panic!("An active sessions needs to be obtained before using Tidal"),
-            },
-        };
-        session_id.to_owned()
+        // Here it's safe to use unwrap because in ::new() we already checked that there's a valid session_info
```

Comment:
If `Tidal` is used as mutable reference, it's still ok, right?

Reply:
@jpalvarezl  mmm in what usecase?